collectors/build/jenkins-audit/src/main/java/com/capitalone/dashboard/util/AuditDataCSVFileCreator.java

```diff
@@ -0,0 +1,275 @@
+ *
+ * @since 09/28/2018
+ */
+@SuppressWarnings("PMD")
+public class AuditDataCSVFileCreator {
+
+    // BEGIN_DATE, END_DATE,CSV_OUTPUT_FILE_PATH values finalized by CSV file creator
+    private static final long BEGIN_DATE = 0;
+    private static final long END_DATE = 0;
+    private static final String CSV_OUTPUT_FILE_PATH = "";
+
+    private static final Set<AuditType> AUDIT_TYPE = new HashSet<>();
+    private static final String[] CSV_HEADER = {"LOB", "ASV", "BAP",
+            "ASV_OWNER", "SERV_OWNER", "AUDIT_TYPE", "COLLECTION_STATUS", "AUDIT_STATUS", "URL", "ERROR"};
+    private static final String[] AUDIT_TYPES = {"LIBRARY_POLICY", "CODE_REVIEW", "TEST_RESULT", "STATIC_SECURITY_ANALYSIS",
+            "PERF_TEST", "CODE_QUALITY", "BUILD_REVIEW"};
+    private static final String[] CODE_REVIEW_FAIL_STATUES = {"PEER_REVIEW_LGTM_ERROR", "PEER_REVIEW_LGTM_SELF_APPROVAL", "PEER_REVIEW_GHR_SELF_APPROVAL",
+            "DIRECT_COMMITS_TO_BASE", "MERGECOMMITER_NOT_FOUND", "PULLREQ_NOT_PEER_REVIEWED", "COLLECTOR_ITEM_ERROR",
+            "NO_COMMIT_FOR_DATE_RANGE", "COMMIT_AFTER_PR_MERGE", "SCM_AUTHOR_LOGIN_INVALID", "REPO_NOT_CONFIGURED", "PENDING_DATA_COLLECTION", "GIT_NO_WORKFLOW",
+            "COMMITAUTHOR_EQ_MERGECOMMITER", "PEER_REVIEW_BY_SERVICEACCOUNT", "COMMITAUTHOR_EQ_SERVICEACCOUNT", "MERGECOMMITER_EQ_SERVICEACCOUNT"};
+
+    private static String lob, asv, bap, asvOwner, serviceOwner, type, collectionStatus, auditStatus, error, url;
+    @Autowired
+    private static DashboardRepository dashboardRepository;
+    @Autowired
+    private static DashboardAuditService dashboardAuditService;
+    @Autowired
+    private static CmdbRepository cmdbRepository;
+    private final Logger LOGGER = LoggerFactory.getLogger(AuditDataCSVFileCreator.class);
+    private List<String> entireCSVData = new ArrayList<>();
+
+    @Autowired
+    public AuditDataCSVFileCreator(DashboardRepository dashboardRepository, DashboardAuditService dashboardAuditService, CmdbRepository cmdbRepository) {
+        this.dashboardRepository = dashboardRepository;
+        this.dashboardAuditService = dashboardAuditService;
+        this.cmdbRepository = cmdbRepository;
+        createCSVFile();
+    }
+
+    private void createCSVFile() {
+        // This findByTimestampAfter can be changed based on the need.
+        //Iterable<Dashboard> recentDashboards = dashboardRepository.findByTimestampAfter(BEGIN_DATE);
+        Iterable<Dashboard> recentDashboards = dashboardRepository.findByTitle("CI304346");
+        AUDIT_TYPE.add(AuditType.ALL);
+        this.getAuditResults(recentDashboards, BEGIN_DATE);
+        if (!(entireCSVData == null && entireCSVData.isEmpty())) {
+            this.writeCSV();
+        }
+    }
+
+    private void writeCSV() {
+        try {
+            LOGGER.info("Writing CSV Data...");
+            Files.write(Paths.get(CSV_OUTPUT_FILE_PATH), String.join("\n", entireCSVData).getBytes());
+        } catch (Exception e) {
```

Comment:
Please don't catch Exception.

Reply:
This is utility class, and not part of the collector data collection flow. This class can be executed independent of any other class dependency.